src/components/fader.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Creates a box around the element (assumed to be the camera's PoV) which can be used for fade-to-black.
+ */
+
+const FADE_DURATION_MS = 750;
+
+AFRAME.registerComponent("fader", {
+  schema: {
+    direction: { type: "string", default: "none" } // "in", "out", or "none"
+  },
+
+  init() {
+    const mesh = new THREE.Mesh(
+      new THREE.BoxGeometry(),
+      new THREE.MeshBasicMaterial({ color: 0x0, side: THREE.BackSide, opacity: 0, transparent: true })
+    );
+    mesh.scale.x = mesh.scale.y = mesh.scale.z = 0.05;
+    mesh.matrixNeedsUpdate = true;
+    this.el.object3D.add(mesh);
+    this.mesh = mesh;
```

Comment:
Was going to suggest starting the mesh invisible, but looks like it would be hidden in the first tick anyway.

Reply:
yeah, that's how I had it originally, but it resulted in it never being drawn. i'm not sure why, probably something having to do with the matrix update loop or something.